src/ValidatorChain.php

```diff
@@ -44,12 +52,12 @@ public function __construct(Storage $storage)
     /**
      * Attach a listener to the session validator chain
      *
-     * @param  string $event
+     * @param  string $eventName
      * @param  callable $callback
      * @param  int $priority
      * @return \Zend\Stdlib\CallbackHandler
      */
-    public function attach($event, $callback = null, $priority = 1)
+    public function attach($eventName, callable $callback, $priority = 1)
```

Comment:
Not true; arrays can be php callables. `[$instance, $methodName]` validates
as callable when:

- `$instance` is an object where `$methodName` is a publicly callable
method, OR
- `$instance` is an object or class name and `$methodName` is a publicly
callable static method.
On Oct 25, 2015 2:30 PM, "Maks3w" <notifications@github.com> wrote:

> In src/ValidatorChain.php
> <https://github.com/zendframework/zend-session/pull/8#discussion_r42949108>
> :
>
> >       * @param  callable $callback
> >       * @param  int $priority
> >       * @return \Zend\Stdlib\CallbackHandler
> >       */
> > -    public function attach($event, $callback = null, $priority = 1)
> > +    public function attach($eventName, callable $callback, $priority = 1)
>
> Note the callable typehint make impossible to be array
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-session/pull/8/files#r42949108>.
>